piptools/scripts/sync.py

```diff
@@ -108,6 +119,26 @@ def cli(
             log.error("ERROR: " + msg)
             sys.exit(2)
 
+    if python_executable:
+        resolved_python_executable = shutil.which(python_executable)
+        if resolved_python_executable is None:
+            msg = "Could not resolve '{}' as valid executable path or alias"
+            log.error(msg.format(python_executable))
+            sys.exit(2)
+
+        # Ensure that target python executable has the right version of pip installed
+        pip_version = get_pip_version_for_python_executable(python_executable)
+        required_pip_specification = get_required_pip_specification()
+        if not required_pip_specification.contains(pip_version):
+            msg = (
+                "Target python executable '{}' has pip version {} installed. "
+                "Version {} is expected."
+            )
+            log.error(
+                msg.format(python_executable, pip_version, required_pip_specification)
+            )
+            sys.exit(2)
```

Comment:
This block looks like it could belong in a dedicated function.

Reply:
thanks, extracted _validate_python_executable method